src/librustc_metadata/index.rs

```diff
@@ -124,12 +120,12 @@ impl Lazy<[Index<'tcx>]> {
     /// DefIndex (if any).
     #[inline(never)]
     pub fn lookup(&self, bytes: &[u8], def_index: DefIndex) -> Option<Lazy<Entry<'tcx>>> {
-        let bytes = &bytes[self.position..];
         debug!("Index::lookup: index={:?} len={:?}",
                def_index,
                self.meta);
 
-        let position = u32::read_from_bytes_at(bytes, 1 + def_index.index());
+        let bytes = &bytes[self.position..][..self.meta * 4];
```

Comment:
This is the same as `&bytes[self.position .. self.position + self.meta * 4]`, right? That's rather confusing way of expressing this `:)`.

Reply:
I find it less confusing than repeating `self.position`, but maybe I should do it in two steps, or just bite the bullet and use the `self.position+` form.

Reply:
It just looks like indexing into a 2-dimensional array. I think it's clearer to use the `&foo[start .. end]` form.